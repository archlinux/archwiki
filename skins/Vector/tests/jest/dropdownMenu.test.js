const { addPortletLinkHandler } = require( '../../resources/skins.vector.js/dropdownMenus.js' );

describe( 'addPortletLinkHandler', () => {

	test( 'Adds a span with icon class to dropdown menus', () => {

		// <li> element is the assumed HTML output of mw.util.addPortlet
		document.body.innerHTML = `
		<ul class="vector-menu vector-menu-dropdown">
			<li class="mw-list-item mw-list-item-js" id="test-id">
				<a href="#test-href">
					<span>
						test link content
					</span>
				</a>
			</li>
		</ul>`;

		const mockPortletItem = document.getElementById( 'test-id' );
		// @ts-ignore: mockPortletItem possibly 'null'.
		addPortletLinkHandler( mockPortletItem, { id: 'test-id' } );
		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'Does not add an icon when noicon class is present', () => {

		// <li> element is the assumed HTML output of mw.util.addPortlet
		document.body.innerHTML = `
		<ul class="vector-menu vector-menu-dropdown vector-menu-dropdown-noicon">
			<li class="mw-list-item mw-list-item-js" id="test-id">
				<a href="#test-href">
					<span>
						test link content
					</span>
				</a>
			</li>
		</ul>`;

		const mockPortletItem = document.getElementById( 'test-id' );
		// @ts-ignore: mockPortletItem possibly 'null'.
		addPortletLinkHandler( mockPortletItem, { id: 'test-id' } );
		expect( document.body.innerHTML ).toMatchSnapshot();
	} );

	test( 'JS portlet should be moved to more menu (#p-cactions) at narrow widths', () => {

		// <li> element is the assumed HTML output of mw.util.addPortlet
		document.body.innerHTML = `
		<div class="mw-article-toolbar-container" style="width:1000px">

			<div id="p-namespaces" style="width:1001px"></div>

			<div id="p-variants"></div>

			<div id="p-cactions">
				<ul>
				</ul>
			</div>

			<ul id="p-views" class="vector-menu vector-menu-dropdown">
				<li class="mw-list-item mw-list-item-js" id="test-id">
					<a href="#test-href">
						<span>
							test link content
						</span>
					</a>
				</li>
			</ul>
		</div>`;

		const mockPortletItem = document.getElementById( 'test-id' );
		// @ts-ignore: mockPortletItem  possibly 'null'.
		addPortletLinkHandler( mockPortletItem, { id: 'test-id' } );
		expect( document.body.innerHTML ).toMatchSnapshot();
	} );
} );
