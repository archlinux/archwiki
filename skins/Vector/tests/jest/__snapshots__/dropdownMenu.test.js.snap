// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addPortletLinkHandler Adds a span with icon class to dropdown menus 1`] = `
"
		<ul class=\\"vector-menu vector-menu-dropdown\\">
			<li class=\\"mw-list-item mw-list-item-js\\" id=\\"test-id\\">
				<a href=\\"#test-href\\"><span class=\\"mw-ui-icon mw-ui-icon-vector-gadget-test-id\\"></span>
					<span>
						test link content
					</span>
				</a>
			</li>
		</ul>"
`;

exports[`addPortletLinkHandler Does not add an icon when noicon class is present 1`] = `
"
		<ul class=\\"vector-menu vector-menu-dropdown vector-menu-dropdown-noicon\\">
			<li class=\\"mw-list-item mw-list-item-js\\" id=\\"test-id\\">
				<a href=\\"#test-href\\">
					<span>
						test link content
					</span>
				</a>
			</li>
		</ul>"
`;

exports[`addPortletLinkHandler JS portlet should be moved to more menu (#p-cactions) at narrow widths 1`] = `
"
		<div class=\\"mw-article-toolbar-container\\" style=\\"width:1000px\\">

			<div id=\\"p-namespaces\\" style=\\"width:1001px\\"></div>

			<div id=\\"p-variants\\"></div>

			<div id=\\"p-cactions\\">
				<ul>
				<li class=\\"mw-list-item mw-list-item-js\\" id=\\"test-id\\">
					<a href=\\"#test-href\\"><span class=\\"mw-ui-icon mw-ui-icon-vector-gadget-test-id\\"></span>
						<span>
							test link content
						</span>
					</a>
				</li></ul>
			</div>

			<ul id=\\"p-views\\" class=\\"vector-menu vector-menu-dropdown\\">
				
			</ul>
		</div>"
`;
